lib/restbase.js

```diff
@@ -143,6 +143,9 @@ RESTBase.prototype.request = function (req) {
 RESTBase.prototype._request = function (req) {
     var self = this;
 
+    req.headers = req.headers || {};
+    req.headers['x-request-id'] = req.headers['x-request-id'] || rbUtil.generateRequestId();
```

Comment:
I don't think this will work properly, as we don't generally forward headers internally. I think it would make more sense to create the request id when creating the restbase child instance, and then attach it to all outgoing requests / use it for logging.